src/internal/m365/collection/drive/collections.go

```diff
@@ -319,17 +319,28 @@ func (c *Collections) Get(
 		numDriveItems := c.NumItems - numPrevItems
 		numPrevItems = c.NumItems
 
-		// Attach an url cache
+		// Attach an url cache to the drive if the number of discovered items is
+		// below the threshold. Attaching cache to larger drives can cause
+		// performance issues since cache delta queries start taking up majority of
+		// the hour the refreshed URLs are valid for.
 		if numDriveItems < urlCacheDriveItemThreshold {
-			logger.Ctx(ictx).Info("adding url cache for drive")
+			logger.Ctx(ictx).Infow(
```

Comment:
nit: maybe we should log the drive ID too? Seems like it would be good to know which we're attaching to
```suggestion
			logger.Ctx(ictx).Infow(
				"cache_drive_id", driveID,
```

Reply:
drive ID is already included in `ictx` on line 256 above.